src/core/agent/services/connectionService.ts

```diff
@@ -137,10 +138,39 @@ class ConnectionService extends AgentService {
       );
       const operation = await this.agent.modules.signify.resolveOobi(url);
       const connectionId = operation.response.i;
-      await this.createConnectionKeriMetadata(connectionId, {
+      const connectionMetadata: any = {
         alias: operation.alias,
         oobi: url,
-      });
+      };
+      if (url.includes("groupId")) {
+        const params = new URLSearchParams(new URL(url).search);
+        const groupId = params.get("groupId") as string;
+        const identifierWithGroupId =
+          await AriesAgent.agent.identifiers.getKeriIdentifierByGroupId(
+            groupId
+          );
+        if (identifierWithGroupId) {
+          connectionMetadata.groupId = groupId;
+        } else {
+          //TODO: Open a pop up to get the display name and colors inputs
+          const displayName = utils.uuid();
```

Comment:
We need to think here about how the UI will provide this. Maybe we can make this function return a type:
```
type OobiScan = { type: ConnectionType.NORMAL } | { type: ConnectionType.MULTI_SIG; groupId: string };
```
(where `ConnectionType` is a new enum)

Or something like this. So that the UI can check if it's multi-sig and then pop up the part to create the identifier.